sdk/storage/azure-storage-file-datalake/azure/storage/file/datalake/_deserialize.py

```diff
@@ -0,0 +1,118 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+from typing import (  # pylint: disable=unused-import
+    Union, Optional, Any, Iterable, Dict, List, Type, Tuple,
+    TYPE_CHECKING
+)
+import logging
+
+from azure.core.pipeline.policies import ContentDecodePolicy
+from azure.core.exceptions import (
+    HttpResponseError,
+    ResourceNotFoundError,
+    ResourceModifiedError,
+    ResourceExistsError,
+    ClientAuthenticationError,
+    DecodeError)
+
+
+if TYPE_CHECKING:
+    from datetime import datetime
+    from azure.core.exceptions import AzureError
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def normalize_headers(headers):
+    normalized = {}
+    for key, value in headers.items():
+        if key.startswith('x-ms-'):
+            key = key[5:]
+        normalized[key.lower().replace('-', '_')] = value
+    return normalized
+
+
+def deserialize_metadata(response, obj, headers):  # pylint: disable=unused-argument
+    raw_metadata = {k: v for k, v in response.headers.items() if k.startswith("x-ms-meta-")}
+    return {k[10:]: v for k, v in raw_metadata.items()}
+
+
+def return_response_headers(response, deserialized, response_headers):  # pylint: disable=unused-argument
+    return normalize_headers(response_headers)
+
+
+def return_headers_and_deserialized_path_list(response, deserialized, response_headers):  # pylint: disable=unused-argument
+    return deserialized.paths if deserialized.paths else {}, normalize_headers(response_headers)
+
+
+
+#
+# def process_storage_error(storage_error):
```

Comment:
> def process_storage_error(storage_error): [](start = 1, length = 42)

what's happening here?

Reply:
want to deserialize error from adls but haven't got time to adding